UnitsNet/BaseUnits.cs

```diff
@@ -0,0 +1,161 @@
+
+namespace UnitsNet
+{
+    /// <summary>
+    ///     Represents the base units for a quantity or 
+    /// </summary>
+    public sealed class BaseUnits
+    {
+        /// <inheritdoc />
+        public BaseUnits(LengthUnit length, MassUnit mass, DurationUnit time, ElectricCurrentUnit current,
+            TemperatureUnit temperature, AmountOfSubstanceUnit amount, LuminousIntensityUnit luminousIntensity)
+        {
+            if(length == LengthUnit.Undefined)
+                throw new ArgumentException("The given length unit is undefined.", nameof(length));
+            else if(mass == MassUnit.Undefined)
+                throw new ArgumentException("The given mass unit is undefined.", nameof(mass));
+            else if(time == DurationUnit.Undefined)
+                throw new ArgumentException("The given time unit is undefined.", nameof(time));
+            else if(current == ElectricCurrentUnit.Undefined)
+                throw new ArgumentException("The given electric current unit is undefined.", nameof(current));
+            else if(temperature == TemperatureUnit.Undefined)
+                throw new ArgumentException("The given temperature unit is undefined.", nameof(temperature));
+            else if(amount == AmountOfSubstanceUnit.Undefined)
+                throw new ArgumentException("The given amount of substance unit is undefined.", nameof(amount));
+            else if(luminousIntensity == LuminousIntensityUnit.Undefined)
+                throw new ArgumentException("The given luminous intensity unit is undefined.", nameof(luminousIntensity));
+
+            Length = length;
+            Mass = mass;
+            Time = time;
+            Current = current;
+            Temperature = temperature;
+            Amount = amount;
+            LuminousIntensity = luminousIntensity;
+        }
+
+        /// <inheritdoc />
+        public override bool Equals(object obj)
+        {
+            if(obj is null || !(obj is BaseUnits))
+                return false;
+
+            var other = (BaseUnits)obj;
+
+            return Length == other.Length &&
+                Mass == other.Mass &&
+                Time == other.Time &&
+                Current == other.Current &&
+                Temperature == other.Temperature &&
+                Amount == other.Amount &&
+                LuminousIntensity == other.LuminousIntensity;
+        }
+
+        /// <inheritdoc />
+        public override int GetHashCode()
+        {
+            return new {Length, Mass, Time, Current, Temperature, Amount, LuminousIntensity}.GetHashCode();
+        }
+
+#if !WINDOWS_UWP
+
+        /// <summary>
+        /// Check if two dimensions are equal.
+        /// </summary>
+        /// <param name="left">Left.</param>
+        /// <param name="right">Right.</param>
+        /// <returns>True if equal.</returns>
+        public static bool operator ==(BaseUnits left, BaseUnits right)
+        {
+            return left?.Equals(right) == true;
```

Comment:
This fails to compare two `null` values I think.
https://stackoverflow.com/a/73732/134761

Reply:
Left hand side null is handled by return left?. The right hand side check is handled in the Equals method:

```csharp
        public override bool Equals(object obj)
        {
            if(obj is null || !(obj is BaseUnits))
                return false;
```

Reply:
I will add a test 😃

Reply:
```
object left = null;
object right = null;
(left?.Equals(right) == true).Dump(); // False in linqpad
```

Reply:
That would not call BaseUnits.Equals  ?

Reply:
Fixed :)